### DISCUSSION TITLE
updated_{variable_name} not firing

### DISCUSSION THREAD

**Discussion by zurtri** [django-commons/django-unicorn#290] [Q&A] [answered]
Hi I have the `updated_` function not firing and I'm wondering what I'm doing wrong:

My template:

```
 <select 
       unicorn:model="new_event_horse" 
       id="new_event_horse"
        class="form-control">

    <option value="">--</option>

    {% for horse in horses %}
        <option value="{{ horse.pk }}">{{ horse.display_name }}</option>
    {% endfor %}

</select>
```

And then I have this function:
```
    def updated_new_event_horse(self, value):
        if self.new_event_horse:
            self.new_event_horse_name = "Updated Horse name"
        else:
            self.new_event_horse_name = None
```

The `new_event_horse_name` variable is not getting updated and I don't think the `updated_new_event_horse()` is even being called.

What am I doing wrong?

Thanks!

**Comment by adamghill** [comment, marked as answer]
And you have a `new_event_horse` field in your component? I was testing in my example with your HTML above and this basic component code:

```python
class TestView(UnicornView):
    horses = [Horse(1, "Horse 1"), Horse(2, "Horse 2")]
    new_event_horse = ""

    def updated_new_event_horse(self, value):
        print(f"updated_new_event_horse value: {value}")
```

and I'm seeing those print statements in my terminal:
![Oct-06-2021 20-37-31](https://user-images.githubusercontent.com/317045/136302658-76caf985-1a80-47c2-a44b-4dcb5c3173e9.gif)
 

What I didn't realize is that the method actually gets called twice (once for the original value and then a second time for the new value, which seems a little weird to me). But, let me know if I can help troubleshoot in another way.

**Comment by zurtri** [comment]
I am a complete fool!

I have two classes in the one file and I had accidentally put the updated_ function in the wrong class.

So embarrassing. 

Thank you so much for your help Adam!

**Reply by adamghill** [reply]
Great! Glad to hear it was an easy fix. 👍
